### PR TITLE
BED-5065: Display better error messages when creating a new user

### PR DESCRIPTION
**Pull request by wes-mil**
## Description

Display more descriptive and better formatted errors when failing to create a new user.

## Motivation and Context

This PR addresses: BED-5065

*Why is this change required? What problem does it solve?*

Previous error message would not indicate to the admin user why creating a new user was failing to be created.

## How Has This Been Tested?

Creating a new user with a duplicate principal name now displays a more descriptive error message.

## Screenshots (optional):

Duplicate principal name errors now display as:
Screenshot from 2024-12-09 14-54-15

Generic errors now display as:
Screenshot from 2024-12-09 14-54-48

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [x] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

File: cmd/api/src/database/helper.go [hunk 1]
```diff
@@ -17,16 +17,24 @@
...
+	"strings"
 )
 
 func CheckError(tx *gorm.DB) error {
 	if errors.Is(tx.Error, gorm.ErrRecordNotFound) {
 		return ErrNotFound
 	}
 
+	if tx.Error != nil {
+		if strings.Contains(tx.Error.Error(), "duplicate key value violates unique constraint \"users_principal_name_key\"") {
```
Comment (mistahj67) [inline cmd/api/src/database/helper.go:33]: We want to keep this function agnostic to any specific db table as it's used across the datalayer. This error handling might be better placed in the db method directly associated with user creation

Comment (mistahj67) [reply cmd/api/src/database/helper.go:33]: Yea in both is likely a better place than here IMO

File: cmd/api/src/database/db.go [hunk 2]
```diff
@@ -44,8 +44,10 @@ const (
 )
 
 var (
-	ErrDuplicateAGName = errors.New("duplicate asset group name")
-	ErrDuplicateAGTag  = errors.New("duplicate asset group tag")
+	ErrDuplicateAGName        = errors.New("duplicate asset group name")
+	ErrDuplicateAGTag         = errors.New("duplicate asset group tag")
+	ErrDuplicateUserPrincipal = errors.New("duplicate user principal name")
+	ErrDuplicateUserEmail     = errors.New("duplicate user email")
```
Comment (mistahj67) [inline cmd/api/src/database/db.go:50]: This is leftover right?
```suggestion
```

**Review by mistahj67 — COMMENTED** [review]
Code LGTM, please have someone test this locally to verify UI changes

**Review by ALCooper12 — APPROVED** [review]
Code LGTM and the UI displays the error message properly 👍🏾

**Pull request merged**